file-formats/cfdf/type/zif_aff_cfdf_v2.intf.abap

```diff
@@ -0,0 +1,573 @@
+INTERFACE zif_aff_cfdf_v2
+  PUBLIC.
+
+  TYPES ty_field_name             TYPE c LENGTH 30.
+  TYPES ty_component_identifier   TYPE c LENGTH 5.
+  TYPES ty_business_context       TYPE c LENGTH 30.
+  TYPES ty_code_value             TYPE c LENGTH 30.
+  TYPES ty_data_subject_id_type   TYPE c LENGTH 10.
+  TYPES ty_type                   TYPE c LENGTH 10.
+  TYPES ty_message_direction      TYPE c LENGTH 3.
+  TYPES ty_field_control_property TYPE c LENGTH 1.
+  TYPES ty_aggregation            TYPE c LENGTH 5.
+  TYPES ty_component_kind         TYPE c LENGTH 11.
+  TYPES:
+    "! <p class="shorttext">Labels and Tooltip</p>
+    "! Labels and tooltip
+    BEGIN OF ty_gs_ui_text,
+      "! <p class="shorttext">Translation Language</p>
+      "! Translation language
+      "! $required
+      language             TYPE sy-langu,
+
+      "! <p class="shorttext">Component Kind</p>
+      "! Component kind
+      "! $required
+      "! $values {@link zif_aff_cfdf_v2.data:co_component_kind}
+      "! $default {@link zif_aff_cfdf_v2.data:co_component_kind.main}
+      component_kind       TYPE ty_component_kind,
+
+      "! <p class="shorttext">Component Field Identifier</p>
+      "! Component field identifier
+      "! $required
+      component_identifier TYPE c                 LENGTH 5,
+
+      "! <p class="shorttext">Field Label</p>
+      "! Field label
+      field_label          TYPE c                 LENGTH 40,
+
+      "! <p class="shorttext">Field Tooltip</p>
+      "! Field tooltip
+      "! $required
+      field_tooltip        TYPE c                 LENGTH 60,
+
+      "! <p class="shorttext">Field Heading</p>
+      "! Field heading
+      "! $required
+      field_heading        TYPE c                 LENGTH 55,
+
+      "! <p class="shorttext">Field Medium Label</p>
+      "! Field medium label
+      field_medium_label   TYPE c                 LENGTH 20,
+
+      "! <p class="shorttext">Field Short Label</p>
+      "! Field short label
+      field_short_label    TYPE c                 LENGTH 10,
+    END OF ty_gs_ui_text.
+  TYPES ty_gt_ui_text TYPE STANDARD TABLE OF ty_gs_ui_text
+        WITH KEY language component_kind component_identifier.
```

Comment:
Isn't language missing in the key, now?

Reply:
> Isn't `language` missing in the key, now?

What do you mean? language is part of the key:
  TYPES ty_gt_ui_text TYPE STANDARD TABLE OF ty_gs_ui_text
        WITH KEY language component_kind component_identifier.

Reply:
Sorry, my fault. I just somehow missed it because `language` is rendered as keyword.